system/system.go

```diff
@@ -153,23 +154,47 @@ func (p *PingOperation) Execute(ctx context.Context, c *internal.Clients) ([]*sp
 
 // SwitchControlProcessorOperation represents the parameters of a SwitchControlProcessor operation.
 type SwitchControlProcessorOperation struct {
-	req *spb.SwitchControlProcessorRequest
+	subcomponentName string
+	isPathSet        bool
+	path             *tpb.Path
 }
 
 // NewSwitchControlProcessorOperation creates an empty SwitchControlProcessorOperation.
 func NewSwitchControlProcessorOperation() *SwitchControlProcessorOperation {
-	return &SwitchControlProcessorOperation{req: &spb.SwitchControlProcessorRequest{}}
+	return &SwitchControlProcessorOperation{}
+}
+
+// PathFromSubcomponentName sets the path for route processor to switch from the subcomponentName.
+func (s *SwitchControlProcessorOperation) PathFromSubcomponentName(n string) *SwitchControlProcessorOperation {
+	s.subcomponentName = n
```

Comment:
delete the subcomponentName and isPathSet fields and instead set the path here

Reply:
Initially created these 2 variables so that it is easy to identify if the user is trying to set the path using both functions and gets unexpected results. But, after thinking over it, it seemed like adding a comment could clarify this instead of returning an error. 

Let me know if that sounds good.